config/core/deployments/controller.yaml

```diff
@@ -47,6 +47,14 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
+          - name: CONTROLLER_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: CONTROLLER_UID
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.uid
```

Comment:
The downward API populates these with Pod values, so this PR is setting up bad owner references and Kubernetes GC will randomly delete the deployments as a result.  We hit a similar issue with bad owner references in the webhook several months ago.

I believe @duglin was also just seeing a similar bad behavior in a separate context, and @markusthoemmes got bitten by this a month or two ago.